packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -0,0 +1,157 @@
+    '/baskets' // No trailing / as it's an aggregate endpoint
+]
+
+const updateBasketQuery = (
+    customerId: string | null,
+    parameters: BasketParameters,
+    newBasket: Basket
+): CacheUpdate => {
+    // The `parameters` received includes the client config and parameters from other endpoints
+    // so we need to exclude unwanted parameters in the query key
+    const getBasketParameters: GetBasketParameters = pick(parameters, [
+        'basketId',
+        'locale',
+        'organizationId',
+        'siteId'
+    ])
+    const basketUpdate = {
+        queryKey: [...getBasketPath(parameters), getBasketParameters] as const
+    }
+
+    // We can only update customer baskets if we have a customer!
+    if (!customerId) return {update: [basketUpdate]}
+
+    // Same elision as done for `getBasket`
+    const getCustomerBasketsParameters: GetCustomerBasketsParameters = {
+        customerId,
+        organizationId: parameters.organizationId,
+        siteId: parameters.siteId
+    }
+    const customerBasketsUpdate = {
+        queryKey: [
+            ...getCustomerBasketsPath(getCustomerBasketsParameters),
+            getCustomerBasketsParameters
+        ] as const,
+        updater: (oldData?: CustomerBasketsResult): CustomerBasketsResult | undefined => {
+            // do not update if response basket is not part of existing customer baskets
+            if (!oldData?.baskets?.some((basket) => basket.basketId === parameters.basketId)) {
+                return undefined
+            }
+            const updatedBaskets = oldData.baskets.map((basket) => {
+                return basket.basketId === parameters.basketId ? newBasket : basket
+            })
+            return {
+                ...oldData,
+                // Shopper Customers and Shopper Baskets have different definitions for the `Basket`
+                // type. (99% similar, but that's not good enough for TypeScript.)
+                // TODO: Remove this type assertion when the RAML specs match.
+                baskets: updatedBaskets as CustomerBasketsResult['baskets']
+            }
+        }
+    }
+    return {update: [basketUpdate, customerBasketsUpdate]}
+}
+
+const invalidateCustomerBasketsQuery = (
+    customerId: string | null,
+    parameters: Omit<GetCustomerBasketsParameters, 'customerId'>
+): CacheUpdate => {
+    if (!customerId) return {}
+    return {
+        invalidate: [
+            and(
+                matchesApiConfig(parameters),
+                matchesPath(getCustomerBasketsPath({...parameters, customerId}))
+            )
+        ]
+    }
+}
+
+const updateBasket = (
+    customerId: string | null,
+    {parameters}: {parameters: BasketParameters},
+    response: Basket
+): CacheUpdate => ({
+    // TODO: We only update the basket from the matching locale; we should also invalidate other locales
+    ...updateBasketQuery(customerId, parameters, response),
+    ...invalidateCustomerBasketsQuery(customerId, parameters)
```

Comment:
We already manually update the baskets in `updateBasketQuery`. I don't think we need to invalidate customerBaskets.